### PR TITLE
Added Java implementation

### PR DESCRIPTION
**Pull request by Seregy**

### CODE REVIEW

File: Java/Match.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+     * @param args command-line arguments, not used
+     */
+    public static void main(final String[] args) {
+        String input1 = "My email is someones@mail.com";
+        String input2 = "My email is not@mail";
+        String regularExpression = "[^\\s]+@[^\\s]+\\.[^\\s]+";
+
+        Pattern pattern = Pattern.compile(regularExpression);
+
+        Matcher matcher1 = pattern.matcher(input1);
```
Comment (lundibundi) [inline Java/Match.java]: To make this more explicit and avoid duplicate code I'd suggest to put all inputs in a collection and then output results with something like
`input: result` or `'input': result`.

File: Java/Split.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    /**
+     * Entry point of the program.
+     *
+     * @param args command-line arguments, not used
+     */
+    public static void main(final String[] args) {
+        String input = "Hello World, here we are!";
+
+        //Using String API
```
Comment (lundibundi) [inline Java/Split.java]: Missing space.

**Review by lundibundi — CHANGES_REQUESTED** [review]
Just a few minor changes.

**Comment by lundibundi** [comment]
Also @tshemsedinov I'd like match, replaceWithStringAPI, replaceWithRegexAPI etc to be separate functions in a single class file, but as we have separate files for JS example I'm not sure if this will be appropriate, what's your opinion on this?

**Comment by tshemsedinov** [comment]
@lundibundi it is not a project but a set of examples, so we prepare them in separate files.
Also @Seregy please rename all files as I named them in JS version: `N-name.java` where N will be a number of example, it is good for sequential demonstration.

**Review by tshemsedinov — CHANGES_REQUESTED** [review]
Rename files

**Comment by Seregy** [comment]
I've renamed files and made the necessary changes.

**Review by lundibundi — APPROVED** [review]
LGTM. Thanks for your effort.

**Review by tshemsedinov — APPROVED** [review]

**Pull request merged**